test/dataframe.jl

```diff
@@ -133,59 +133,93 @@ module TestDataFrame
-        @test all(ismissing, df[:, 2])
-        @test all(ismissing, df[:, 3])
+        @test typeof(df[1]) == Vector{Union{Int, Missing}}
+        @test typeof(df[2]) == Vector{Union{Float64, Missing}}
+        @test typeof(df[3]) == Vector{Union{String, Missing}}
+        @test all(ismissing, df[1])
+        @test all(ismissing, df[2])
+        @test all(ismissing, df[3])
+        # TODO: enable those tests after getindex deprecation period
+        # @test typeof(df[:, 1]) == Vector{Union{Int, Missing}}
+        # @test typeof(df[:, 2]) == Vector{Union{Float64, Missing}}
+        # @test typeof(df[:, 3]) == Vector{Union{String, Missing}}
+        # @test all(ismissing, df[:, 1])
+        # @test all(ismissing, df[:, 2])
+        # @test all(ismissing, df[:, 3])
 
         df = DataFrame([Union{Int, Missing}, Union{Float64, Missing}, Union{String, Missing}],
                     [:A, :B, :C], 100)
         @test size(df, 1) == 100
         @test size(df, 2) == 3
-        @test typeof(df[:, 1]) == Vector{Union{Int, Missing}}
-        @test typeof(df[:, 2]) == Vector{Union{Float64, Missing}}
-        @test typeof(df[:, 3]) == Vector{Union{String, Missing}}
-        @test all(ismissing, df[:, 1])
-        @test all(ismissing, df[:, 2])
-        @test all(ismissing, df[:, 3])
+        @test typeof(df[1]) == Vector{Union{Int, Missing}}
+        @test typeof(df[2]) == Vector{Union{Float64, Missing}}
+        @test typeof(df[3]) == Vector{Union{String, Missing}}
+        @test all(ismissing, df[1])
+        @test all(ismissing, df[2])
+        @test all(ismissing, df[3])
+        # TODO: enable those tests after getindex deprecation period
+        # @test typeof(df[:, 1]) == Vector{Union{Int, Missing}}
+        # @test typeof(df[:, 2]) == Vector{Union{Float64, Missing}}
+        # @test typeof(df[:, 3]) == Vector{Union{String, Missing}}
+        # @test all(ismissing, df[:, 1])
+        # @test all(ismissing, df[:, 2])
+        # @test all(ismissing, df[:, 3])
 
         df = DataFrame([Union{Int, Missing}, Union{Float64, Missing}, Union{String, Missing}],
                     [:A, :B, :C], [false, false, true], 100)
         @test size(df, 1) == 100
         @test size(df, 2) == 3
-        @test typeof(df[:, 1]) == Vector{Union{Int, Missing}}
-        @test typeof(df[:, 2]) == Vector{Union{Float64, Missing}}
-        @test typeof(df[:, 3]) <: CategoricalVector{Union{String, Missing}}
-        @test all(ismissing, df[:, 1])
-        @test all(ismissing, df[:, 2])
-        @test all(ismissing, df[:, 3])
+        @test typeof(df[1]) == Vector{Union{Int, Missing}}
+        @test typeof(df[2]) == Vector{Union{Float64, Missing}}
+        @test typeof(df[3]) <: CategoricalVector{Union{String, Missing}}
+        @test all(ismissing, df[1])
+        @test all(ismissing, df[2])
+        @test all(ismissing, df[3])
+        # TODO: enable those tests after getindex deprecation period
+        # @test typeof(df[:, 1]) == Vector{Union{Int, Missing}}
```

Comment:
Can you use `@test_broken` for these?

Reply:
This is not broken - the test passes, but prints warnings (massively) and I understood that you wanted to avoid printing unnecessary warnings.

Reply:
I think that's OK if the tests print warnings, as long as they are correct in the long term. Better have good tests that are temporarily verbose than having to enable them manually later.

Reply:
So I understand we go the following route:
* it is not a problem if tests print warnings;
* however, we want to avoid printing warnings by library functions in normal usage.

Yes?

Reply:
Right.